### PR TITLE
add statistic model

### PR DESCRIPTION
**Pull request by lxs-turing**
## Description of the change
add statistic model
// TODO Description here

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

// TODO Fix #...

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: stats/models.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from datetime import datetime
+from django.utils.functional import cached_property
+from django.db import models
+
+from turing.common.django.db import TimedModel
+
+
+class CaseStatistics(TimedModel):
+    customer_id = models.CharField(max_length=32)
```
Comment (longshine) [inline stats/models.py:9]: Change to `org_code` to keep same as other services.

File: stats/apps.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class StatsConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
```
Comment (longshine) [inline stats/apps.py:5]: This line is not needed since you've already set it in settings.py.

File: stats/models.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+from datetime import datetime
+from django.utils.functional import cached_property
+from django.db import models
+
+from turing.common.django.db import TimedModel
+
+
+class CaseStatistics(TimedModel):
```
Comment (longshine) [inline stats/models.py:8]: Rename to `AlertStatistic`

File: stats/models.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    customer_id = models.CharField(max_length=32)
+    site_id = models.IntegerField()
+    box_id = models.CharField(max_length=32)
+    camera_id = models.IntegerField()
+    algo_code = models.CharField(max_length=64)
```
Comment (longshine) [inline stats/models.py:13]: Change to `event_type`

File: stats/models.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    scenario_code = models.CharField(max_length=64)
+    eventtype_code = models.CharField(max_length=1023)
+    total_case = models.PositiveIntegerField(default=0)
+    total_false_case = models.PositiveIntegerField(default=0)
+    total_pending_case = models.PositiveIntegerField(default=0)
+    total_read_case = models.PositiveIntegerField(default=0)
+
+    class Meta:
+        db_table = 'ka_stats_case_statistics'
```
Comment (longshine) [inline stats/models.py:22]: Remove `db_table`. Let's use the generated table name by django.

File: stats/models.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+
+class CaseStatistics(TimedModel):
+    customer_id = models.CharField(max_length=32)
+    site_id = models.IntegerField()
+    box_id = models.CharField(max_length=32)
+    camera_id = models.IntegerField()
+    algo_code = models.CharField(max_length=64)
+    scenario_code = models.CharField(max_length=64)
+    eventtype_code = models.CharField(max_length=1023)
```
Comment (longshine) [inline stats/models.py:15]: What are these two fields for? I did not see them in `ka_service`.

File: stats/models.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    total_case = models.PositiveIntegerField(default=0)
+    total_false_case = models.PositiveIntegerField(default=0)
+    total_pending_case = models.PositiveIntegerField(default=0)
+    total_read_case = models.PositiveIntegerField(default=0)
```
Comment (longshine) [inline stats/models.py:19]: What is this field for? I did not see it in `ka_service`.

File: stats/models.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+    class Meta:
+        db_table = 'ka_stats_case_statistics'
+        constraints = [
+            models.UniqueConstraint(
+                name='idx_unique_case_statistics',
+                fields=('customer_id', 'site_id', 'camera_id', 'eventtype_code')),
+        ]
```
Comment (longshine) [inline stats/models.py:27]: Ref: [unique_together](https://docs.djangoproject.com/en/4.0/ref/models/options/#unique-together)

File: stats/models.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        indexes = [
+            models.Index(
+                fields=('customer_id', 'eventtype_code'),
+                name='idx_ka_stats_customer_eventtype'
+            ),
+        ]
```
Comment (longshine) [inline stats/models.py:33]: Ref: https://docs.djangoproject.com/en/4.0/ref/models/options/#index-together

File: stats/models.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+
+class SiteStatistics(TimedModel):
+    customer_id = models.CharField(max_length=32)
+    site_id = models.IntegerField()
+    total_box = models.PositiveIntegerField(default=0)
+    total_box_offline = models.PositiveIntegerField(default=0)
```
Comment (longshine) [inline stats/models.py:40]: I did not find this in ka_service. Where is it from?

File: stats/models.py [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+        ]
+        indexes = [
+            models.Index(
+                fields=('customer_id', 'site_id'),
+                name='idx_ka_stats_customer_site'
+            ),
+        ]
+
+
+class SiteCaseHourTrend(models.Model):
```
Comment (longshine) [inline stats/models.py:64]: Change to `SiteAlertHourTrend`

File: stats/models.py [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+    customer_id = models.CharField(max_length=32)
```
Comment (longshine) [inline stats/models.py:65]: `org_code`

File: stats/models.py [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+    site_id = models.IntegerField()
+    hour = models.PositiveIntegerField()
+    total_new_case = models.PositiveIntegerField(default=0)
```
Comment (longshine) [inline stats/models.py:68]: `total_new_alerts`

**Review by longshine — CHANGES_REQUESTED** [review]

File: stats/models.py (hunk 4 above)
Comment (lxs-turing) [reply stats/models.py:13]: algo_code != event_type,  algo contain many eventtypes, wentao zhao bind the relationship between algo_code and event_type in ka_algo, i think event_type is the smallest statistical unit.

File: stats/models.py (hunk 10 above)
Comment (lxs-turing) [reply stats/models.py:40]: Keep naming style consistent, i modify box_count to total_box, box_offline_count to total_box_offline

File: stats/models.py (hunk 9 above)
Comment (lxs-turing) [reply stats/models.py:33]: Use the [indexes](https://docs.djangoproject.com/en/4.0/ref/models/options/#django.db.models.Options.indexes) option instead.

The newer indexes option provides more functionality than index_together. index_together may be deprecated in the future.

Comment (longshine) [reply stats/models.py]: Oh I see. Suggest to remove the name then. Let's use the name generated by django.

File: stats/models.py (hunk 4 above)
Comment (longshine) [reply stats/models.py:13]: Then why is here a `algo_code`? I did not see it in ka_service.

Comment (lxs-turing) [reply stats/models.py:13]: algo_code is not used at present, which is convenient for future use.

File: stats/models.py [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+    total_false_alert = models.PositiveIntegerField(default=0)
+    total_floor = models.IntegerField(default=0)
+    total_building = models.IntegerField(default=0)
+    total_alert_type = models.PositiveIntegerField(default=0)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(name='idx_unique_site_statistics', fields=('org_code', 'site_id'))
+        ]
+        indexes = [
```
Comment (longshine) [inline stats/models.py:51]: unique_together

File: stats/models.py [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+            models.Index(fields=['org_code', 'site_id']),
+        ]
+
+
+class SiteAlertHourTrend(models.Model):
+    org_code = models.CharField(max_length=32)
+    site_id = models.IntegerField()
```
Comment (longshine) [inline stats/models.py:58]: BigIntegerField

**Review by longshine — APPROVED** [review]

**Review by longshine — APPROVED** [review]

**Pull request merged**